src/table/utils/accessibility-utils.ts

```diff
@@ -34,44 +36,96 @@ export const updateFocus = ({ focusType, cell }: CellFocusProps) => {
 };
 
-export const removeAndFocus = (
-  newCoord: [number, number],
+/**
+ * Calculates the next cell to focus
+ */
+export const getNextCellCoord = (
+  evt: React.KeyboardEvent,
   rootElement: HTMLElement,
-  setFocusedCellCoord: React.Dispatch<React.SetStateAction<[number, number]>>,
-  keyboard: stardust.Keyboard
-) => {
-  updateFocus({ focusType: 'removeTab', cell: findCellWithTabStop(rootElement) });
-  setFocusedCellCoord(newCoord);
-  keyboard.enabled && keyboard.focus?.();
+  cellCoord: [number, number],
+  allowedRows: {
+    top: number;
+    bottom: number;
+  } = { top: 0, bottom: 0 }
+): [number, number] => {
+  const rowCount = rootElement.getElementsByClassName('sn-table-row').length;
+  const columnCount = rootElement.getElementsByClassName('sn-table-head-cell').length;
+  let [nextRow, nextCol] = cellCoord;
+
+  switch (evt.key) {
+    case 'ArrowDown':
+      nextRow < rowCount - 1 - allowedRows.bottom && nextRow++;
+      break;
+    case 'ArrowUp':
+      nextRow > allowedRows.top && nextRow--;
+      break;
+    case 'ArrowRight':
+      // allowedRows.top greater than 0 means we are in selection mode
+      if (allowedRows.top > 0) break;
+      if (nextCol < columnCount - 1) {
+        nextCol++;
+      } else if (nextRow < rowCount - 1) {
+        nextRow++;
+        nextCol = 0;
+      }
+      break;
+    case 'ArrowLeft':
+      // allowedRows.top greater than 0 means we are in selection mode
+      if (allowedRows.top > 0) break;
+      if (nextCol > 0) {
+        nextCol--;
+      } else if (nextRow > 0) {
+        nextRow--;
+        nextCol = columnCount - 1;
+      }
+      break;
+    default:
+      break;
+  }
+
+  return [nextRow, nextCol];
 };
 
-export const handleClickToFocusBody = (
-  cell: Cell,
+/**
+ * Resets and adds new focus to a table cell based which key is pressed
+ */
+export const moveFocus = (
+  evt: React.KeyboardEvent,
   rootElement: HTMLElement,
+  cellCoord: [number, number],
   setFocusedCellCoord: React.Dispatch<React.SetStateAction<[number, number]>>,
-  keyboard: stardust.Keyboard,
-  totalsPosition: TotalsPosition
+  allowedRows?: {
+    top: number;
+    bottom: number;
+  }
 ) => {
-  const { rawRowIdx, rawColIdx } = cell;
-  const adjustedRowIdx = totalsPosition === 'top' ? rawRowIdx + 2 : rawRowIdx + 1;
-  removeAndFocus([adjustedRowIdx, rawColIdx], rootElement, setFocusedCellCoord, keyboard);
+  updateFocus({ focusType: 'removeTab', cell: evt.target as HTMLTableCellElement });
+  const nextCellCoord = getNextCellCoord(evt, rootElement, cellCoord, allowedRows);
+  const nextCell = getCellElement(rootElement, nextCellCoord);
+  updateFocus({ focusType: 'focus', cell: nextCell });
+  setFocusedCellCoord(nextCellCoord);
+
+  return nextCell;
 };
 
-export const handleClickToFocusHead = (
-  columnIndex: number,
+/**
+ * Resets and adds new focus to the cell at position newCoord. no need for element.focus() since that is done natively
+ */
+export const removeTabAndFocusCell = (
```

Comment:
Thanks for adding comments, the focus related functions can be confusing sometimes.